tests/integration/hive_metastore/test_pipeline_migrate.py

```diff
@@ -0,0 +1,158 @@
+from datetime import timedelta
+
+from databricks.sdk.service.jobs import Task, PipelineTask
+from databricks.sdk.service.pipelines import NotebookLibrary, PipelineLibrary, PipelineCluster, PipelineState
+from databricks.sdk.service.workspace import Language
+
+from databricks.labs.ucx.hive_metastore.pipelines_migrate import PipelinesMigrator
+
+from ..assessment.test_assessment import _PIPELINE_CONF
+
+_TEST_STORAGE_ACCOUNT = "storage_acct_1"
+_TEST_TENANT_ID = "directory_12345"
+
+
+def test_pipeline_migrate(
+    ws, make_pipeline, make_random, watchdog_purge_suffix, make_directory, runtime_ctx, make_mounted_location
+) -> None:
+    src_schema = runtime_ctx.make_schema(catalog_name="hive_metastore")
+
+    src_tables = [
+        runtime_ctx.make_table(catalog_name="hive_metastore", schema_name=src_schema.name, non_delta=True),
+        runtime_ctx.make_table(
+            schema_name=src_schema.name,
+            external_csv=make_mounted_location,
+            columns=[("`foobar`", "STRING")],
+        ),
+    ]
+
+    dlt_notebooks = [
+        runtime_ctx.make_notebook(
+            content=f"""create or refresh streaming table st1\nas select * from stream(hive_metastore.{src_schema.name}.{src_tables[0].name})""".encode(
+                "ASCII"
+            ),
+            path=f"{make_directory()}/dlt_notebook_1",
+            language=Language.SQL,
+        ),
+        runtime_ctx.make_notebook(
+            content=f"""create or refresh streaming table st2\nas select * from stream(hive_metastore.{src_schema.name}.{src_tables[1].name})""".encode(
+                "ASCII"
+            ),
+            path=f"{make_directory()}/dlt_notebook_2",
+            language=Language.SQL,
+        ),
+    ]
+
+    pipeline_names = [
+        f"pipeline-{make_random(4).lower()}-{watchdog_purge_suffix}",
+        f"pipeline-{make_random(4).lower()}-{watchdog_purge_suffix}",
+        f"skip-pipeline-{make_random(4).lower()}-{watchdog_purge_suffix}",
+    ]
+    created_pipelines = [
+        make_pipeline(
+            configuration=_PIPELINE_CONF,
+            name=pipeline_names[0],
+            target=runtime_ctx.make_schema(catalog_name="hive_metastore").name,
+            libraries=[PipelineLibrary(notebook=NotebookLibrary(path=str(dlt_notebooks[0])))],
+            clusters=[
+                PipelineCluster(
+                    node_type_id=ws.clusters.select_node_type(local_disk=False, min_memory_gb=16),
+                    label="default",
+                    num_workers=1,
+                    custom_tags={"cluster_type": "default", "RemoveAfter": watchdog_purge_suffix},
+                )
+            ],
+        ),
+        make_pipeline(
```

Comment:
Consider compacting and reduce number of assertments.

Reply:
Yes, if the integration test does not take too long to execute

Reply:
removed the part where pipeline updates are started and asserted until they are idle as this takes long.